targeteditor/src/main/scala/explore/targeteditor/CoordinatesForm.scala

```diff
@@ -29,10 +31,14 @@ import react.semanticui.widths._
 
 final case class CoordinatesForm(
   target:           SiderealTarget,
-  searchAndGo:      String => Callback,
+  searchAndGo:      NonEmptyString => Callback,
   goToRaDec:        Coordinates => Callback
 )(implicit val ctx: AppContextIO)
-    extends ReactProps[CoordinatesForm](CoordinatesForm.component)
+    extends ReactProps[CoordinatesForm](CoordinatesForm.component) {
+  def submit(searchTerm: String): Callback =
+    refineV[NonEmpty](searchTerm)
+      .fold(_ => Callback.empty, s => searchAndGo(s).when(s =!= target.name).asCBO)
```

Comment:
Why is `.asCBO` needed instead of `.void`?

Reply:
When doing `when` you get a `CallbackOption` that's why `asCBO` but now I realized I can also use `when_` to get directly a `Callback`